packages/shared/hooks/useVault.ts

```diff
@@ -0,0 +1,220 @@
+import { useEffect, useState } from 'react'
+import { useWallet } from './useWallet'
+import { ethers, FixedNumber } from 'ethers'
```

Comment:
Can't we handle BigInt instead?
I don't want to add unnecessary dependency.

Reply:
fixed with viem instead ethers at https://github.com/mycel-labs/frontend/pull/17/commits/698616916cbe9ce48c844e6dabfabc51f7c5659f